includes/QueryPage.php

```diff
@@ -443,11 +443,44 @@ public function getCachedTimestamp() {
 	}
 
 	/**
+	 * Get max number of results we can return.
+	 *
+	 * Most QueryPage subclasses use $wgQueryCacheLimit as a default value, but this can be customized.
+	 *
+	 * @author macbre
+	 * @see SUS-1675
+	 *
+	 * @return int
+	 */
+	protected function getMaxResults() {
+		global $wgQueryCacheLimit, $wgQueryPages;
+
+		// that's our default return value
+		$limit = $wgQueryCacheLimit;
+
+		// get the $wgQueryPages entry for the current QueryPage class
+		$currentQueryPage = get_class( $this );
+		$entries = array_values( array_filter(
+			$wgQueryPages,
+			function( array $item ) use ( $currentQueryPage ) {
+				// class name should match
+				return $item[0] === $currentQueryPage;
+			}
+		) );
+
+		if ( is_array( $entries ) && !empty( $entries[0][2] ) ) {
+			$limit = $entries[0][2];
+		}
+
+		return $limit;
+	}
+
+	/**
 	 * This is the actual workhorse. It does everything needed to make a
 	 * real, honest-to-gosh query page.
 	 */
 	function execute( $par ) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)